includes/class-wc-payments-order-success-page.php

```diff
@@ -188,15 +188,22 @@ public function show_woocommerce_payments_payment_method_name( $payment_method_t
 			return $this->show_woopay_payment_method_name( $order );
 		}
 
+		// Check if this is an Express Checkout payment (Google Pay, Apple Pay, etc.).
+		$express_checkout_payment_method = $order->get_meta( '_wcpay_express_checkout_payment_method' );
+		if ( ! empty( $express_checkout_payment_method ) ) {
+			return $this->show_express_checkout_payment_method_name( $order, $express_checkout_payment_method );
+		}
+
 		$gateway = WC()->payment_gateways()->payment_gateways()[ $payment_method_id ];
 
 		if ( ! is_object( $gateway ) || ! method_exists( $gateway, 'get_payment_method' ) ) {
 			return $payment_method_title;
 		}
 
 		$payment_method = $gateway->get_payment_method( $order );
-		// GooglePay/ApplePay/Link/Card to be supported later.
```

Comment:
In reality, this was just showing `Card` for Google Pay/Apple Pay orders.
I ensured that it now shows `Google Pay ending in 4242` or `Apple Pay ending in 4242`.